pkg/providers/amifamily/ami.go

```diff
@@ -132,6 +132,38 @@ func (p *Provider) Get(ctx context.Context, nodeTemplate *v1alpha1.AWSNodeTempla
 	return amiIDs, nil
 }
 
+func (p *Provider) GetAMIWithRequirements(ctx context.Context, nodeTemplate *v1alpha1.AWSNodeTemplate, instanceTypes []*cloudprovider.InstanceType, amiFamily AMIFamily) (map[AMI]scheduling.Requirements, error) {
+	kubernetesVersion, err := p.KubeServerVersion(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("getting kubernetes version %w", err)
+	}
+	amiIDs := map[string][]*cloudprovider.InstanceType{}
+	amiRequirements, err := p.getAMIRequirements(ctx, nodeTemplate)
+	if err != nil {
+		return nil, err
+	}
+	if len(amiRequirements) == 0 {
+		for _, instanceType := range instanceTypes {
+			amiID, err := p.getDefaultAMIFromSSM(ctx, amiFamily.SSMAlias(kubernetesVersion, instanceType))
+			if err != nil {
+				return nil, err
+			}
+			amiIDs[amiID] = append(amiIDs[amiID], instanceType)
+		}
+		amifilter := map[string]string{"aws-ids": strings.Join(lo.Keys(amiIDs), ",")}
+		amiRequirements, err = p.selectAMIs(ctx, amifilter)
```

Comment:
This requirement resolution doesn't work in the case of the Nvidia GPU case. We need a special additional requirement for the AMD GPU case, otherwise it will just be AMD64 like the other one